ruby/deepgram-audio-summary/index.rb

```diff
@@ -0,0 +1,42 @@
+require "uri"
+require "net/http"
+require "openssl"
+require "json"
+
+
+def main(req, res)
+  fileUrl = nil
+  begin
+    payload = JSON.parse(req.payload)
+    fileUrl = payload["fileUrl"]
+  rescue Exception => err
+      puts err
+      raise "Payload is invalid."
+  end
+  
+  url = URI("https://api.deepgram.com/v1/listen?summarize=true&punctuate=true")
+  http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+  request = Net::HTTP::Post.new(url)
+  request["content-type"] = "application/json"
+  request["Authorization"] = "Token #{req.env["DEEPGRAM_API_KEY"]}"
+  request.body = "{\"url\":\"#{fileUrl}\"}"
+
+  begin
+    response = http.request(request)
+  rescue Exception => err
+    res.json({
+      success: false,
+      deepgramData: "Please provide a valid file URL."
```

Comment:
Just because an error occurs, doesn't mean the URL was invalid. It would be best to return the response body from deepgram here as `message` rather than `deepgramData`